src/Build/BackEnd/Components/ResourceManager/ResourceManagerService.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Shared;
+using System.Threading;
+
+#nullable enable
+
+namespace Microsoft.Build.BackEnd.Components.ResourceManager
+{
+    class ResourceManagerService : IBuildComponent
+    {
+        Semaphore? s = null;
+
+#if DEBUG
+        public int TotalNumberHeld = -1;
+        public string? SemaphoreName;
+#endif
+
+        internal static IBuildComponent CreateComponent(BuildComponentType type)
+        {
+            ErrorUtilities.VerifyThrow(type == BuildComponentType.TaskResourceManager, "Cannot create components of type {0}", type);
+
+            return new ResourceManagerService();
+        }
+
+        public void InitializeComponent(IBuildComponentHost host)
+        {
+            string semaphoreName = host.BuildParameters.ResourceManagerSemaphoreName;
```

Comment:
Do you think it would be worth checking at the start to see if it's already been initialized?